### PR TITLE
TST: consolidate version checks, remove outdated

### LINKED ISSUE #1063: Remove version compatibility code
We can remove some old functions that are no longer needed, etc.. A lot of the time there are source comments so they're easy to find with grep. For example, #127.

Will need to do similar for pandas workarounds 0.7.x -> 0.8.x, mainly in tsa.

### PR DESCRIPTION
**Pull request by jbrockmendel**
Closes #1063.

We recently removed some version checks for unsupported numpy/scipy/pandas versions, but missed a few.  This gets most or all of the remaining ones.

Consolidates version checks in the compat files so that in the future they are all in one place when it comes time to update the minimum required versions.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Related: #5036

File: statsmodels/compat/pandas.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 
 version = LooseVersion(pandas.__version__)
+pandas_lt_0_19_2 = version < '0.19.1'
```
[the remark this answers is not in the document]
Comment (bashtage) [reply statsmodels/compat/pandas.py]: Surely should be lte and <= '0.19.2'

**Comment by bashtage** [comment]
LGTM

**Comment by jbrockmendel** [comment]
@josef-pkt defacto green, owed by bashtage, and no merge conflicts

**Comment by jbrockmendel** [comment]
@josef-pkt merge conflicts are going to start piling up before too long...

**Comment by jbrockmendel** [comment]
@josef-pkt this consolidation should definitely happen before 0.10.0

**Comment by jbrockmendel** [comment]
@josef-pkt any thoughts here?

**Comment by jbrockmendel** [comment]
@josef-pkt let's do version-checking right

**Comment by jbrockmendel** [comment]
@ChadFulton prevent future complications

**Pull request closed without merging**